evidence/pool.go

```diff
@@ -9,22 +9,24 @@ import (
 // EvidencePool maintains a pool of valid evidence
 // in an EvidenceStore.
 type EvidencePool struct {
-	params types.EvidenceParams
 	logger log.Logger
 
-	state         types.State // TODO: update this on commit!
 	evidenceStore *EvidenceStore
 
+	chainID         string
+	lastBlockHeight int64
+	params          types.EvidenceParams
+
 	// never close
 	evidenceChan chan types.Evidence
 }
 
-func NewEvidencePool(params types.EvidenceParams, evidenceStore *EvidenceStore, state types.State) *EvidencePool {
+func NewEvidencePool(params types.EvidenceParams, evidenceStore *EvidenceStore, state *types.State) *EvidencePool {
```

Comment:
Why are we passing a pointer and copying?  Might as well just not receive a pointer...

Reply:
why are we making a copy? and why not `state.Copy()`?